### PR TITLE
Add aten::sigmoid, aten::sigmoid_backward and their variants

### PR DESCRIPTION
**Pull request by AlienLiang23**

### CODE REVIEW

**Comment by AlienLiang23** [comment]
image
all cases passed except **TestCommonXPU.test_out_warning_nn_functional_logsigmoid_xpu** which is already in the skip list

**Review by xytintel — APPROVED** [review]

File: src/aten/sycl/BinaryKernels.cpp [hunk 1]
```diff
@@ -74,6 +74,41 @@ void div_kernel(TensorIteratorBase& iter) {
...
+    using opmath_t = at::opmath_type<scalar_t>;
+    const auto one = opmath_t{1.};
+    const auto comp_b = static_cast<opmath_t>(b);
+    const auto comp_a = static_cast<opmath_t>(a);
+    return static_cast<scalar_t>(comp_a * std::conj((one - comp_b) * comp_b));
+  }
+};
+
+template <typename scalar_t>
+struct SigmoidBackwardFunctor2 {
```
Comment (fengyuan14) [inline src/aten/sycl/BinaryKernels.cpp:89]: Please give meaningful naming.

Comment (fengyuan14) [reply src/aten/sycl/BinaryKernels.cpp:89]: Like `SigmoidBackwardFloatingFunctor`

File: src/aten/sycl/BinaryKernels.cpp [hunk 2]
```diff
@@ -74,6 +74,41 @@ void div_kernel(TensorIteratorBase& iter) {
   }
 }
 
+template <typename scalar_t>
+struct SigmoidBackwardFunctor {
```
Comment (fengyuan14) [inline src/aten/sycl/BinaryKernels.cpp:78]: Please give meaningful naming.

File: src/aten/sycl/BinaryKernels.cpp [hunk 3]
```diff
@@ -74,6 +74,41 @@ void div_kernel(TensorIteratorBase& iter) {
   }
 }
 
+template <typename scalar_t>
```
Comment (fengyuan14) [inline src/aten/sycl/BinaryKernels.cpp:77]: Please move the change to corresponding file as stock CUDA, It should be `aten/src/ATen/native/cuda/BinaryMiscBackwardOpsKernels.cu`?

File: src/aten/sycl/BinaryKernels.cpp [hunk 4]
```diff
@@ -74,6 +74,41 @@ void div_kernel(TensorIteratorBase& iter) {
...
+    AT_DISPATCH_COMPLEX_TYPES_AND(
+        kComplexHalf, dtype, "sigmoid_backward_xpu", [&]() {
+          gpu_kernel(iter, SigmoidBackwardFunctor<scalar_t>());
+        });
+  } else {
+    AT_DISPATCH_FLOATING_TYPES_AND2(
+        at::ScalarType::Half,
+        at::ScalarType::BFloat16,
+        dtype,
+        "sigmoid_backward_cuda",
```
Comment (dvrogozh) [inline src/aten/sycl/BinaryKernels.cpp:107]: typo?
```suggestion
        "sigmoid_backward_xpu",
```

**Review by dvrogozh — CHANGES_REQUESTED** [review]

File: src/aten/sycl/BinaryKernels.cpp (hunk 4 above)
Comment (fengyuan14) [reply src/aten/sycl/BinaryKernels.cpp]: Fixed by the file refactor.

**Pull request merged**